game/shared/gamesystems/spawngroup_manager.h

```diff
@@ -0,0 +1,371 @@
+	virtual bool IsLevelTransition() const = 0;
+	virtual const char *GetLevelTransitionPreviousMap() const = 0;
+	virtual const char *GetLevelTransitionLandmarkName() const = 0;
+	virtual CKeyValues3Context *GetEntityKeyValuesAllocator() = 0;
+	virtual CEntityInstance *CreateEntityToSpawn(SpawnGroupHandle_t hSpawnGroup, const matrix3x4a_t &vecSpawnOffset, CreateSpawnGroupType_t createType, const CEntityKeyValues *pEntityKeyValues) = 0;
+	virtual void SpawnEntities() = 0;
+	virtual void Release() = 0;
+};
+
+class IGameSpawnGroupMgr
+{
+public:
+	virtual void AllocateSpawnGroup(SpawnGroupHandle_t h, ISpawnGroup *pSpawnGroup) = 0;
+	virtual void ReleaseSpawnGroup(SpawnGroupHandle_t h) = 0;
+	virtual void SpawnGroupInit(SpawnGroupHandle_t handle, IEntityResourceManifest *pManifest, IEntityPrecacheConfiguration *pConfig, ISpawnGroupPrerequisiteRegistry *pRegistry) = 0;
+	virtual ILoadingSpawnGroup *CreateLoadingSpawnGroup(SpawnGroupHandle_t handle, bool bSynchronouslySpawnEntities, bool bConfirmResourcesLoaded, const CUtlVector<const CEntityKeyValues *> *pKeyValues) = 0;
+	virtual ILoadingSpawnGroup *CreateLoadingSpawnGroupForSaveFile(SpawnGroupHandle_t handle, const void *pSaveData, size_t nSaveDataSize) = 0;
+	virtual void SpawnGroupSpawnEntities(SpawnGroupHandle_t handle) = 0;
+	virtual void SpawnGroupShutdown(SpawnGroupHandle_t handle) = 0;
+	virtual void SetActiveSpawnGroup(SpawnGroupHandle_t handle) = 0;
+	virtual void UnloadSpawnGroup(SpawnGroupHandle_t hSpawnGroup, bool bSaveEntities) = 0;
+	virtual SpawnGroupHandle_t GetParentSpawnGroup(SpawnGroupHandle_t hSpawnGroup) = 0;
+	virtual int GetChildSpawnGroupCount(SpawnGroupHandle_t handle) = 0;
+	virtual SaveRestoreDataHandle_t SaveGame_Start(const char *pSaveName, const char *pOldLevel, const char *pszLandmarkName) = 0;
+	virtual bool SaveGame_Finalize(SaveRestoreDataHandle_t hSaveRestore) = 0;
+	virtual bool FrameUpdatePostEntityThink(const EventServerPostEntityThink_t &msg) = 0;
+	virtual const char *SaveGame_GetLastSaveFile(SaveRestoreDataHandle_t hSaveRestore) = 0;
+	virtual bool IsGameReadyToSave() = 0;
+	virtual unsigned long UnkGetGlobal1() = 0;
+	virtual void * const UnkGetGameEntitySystemSync1() = 0;
+	virtual void UnkRelease1() = 0;
+	virtual void UnkRelease2() = 0;
+	virtual void YouForgot_DECLARE_GAME_SYSTEM_InYourClassDefinition() = 0;
+	virtual void UnkSpawnGroupManagerGameSystemMember1() = 0;
+	virtual void GameInit(const EventGameInit_t &msg) = 0;
+	virtual void GameShutdown(const EventGameInit_t &msg) = 0;
+	virtual void FrameBoundary(const EventGameInit_t &msg) = 0;
+	virtual void PreSpawnGroupLoad(const EventPreSpawnGroupLoad_t &msg) = 0;
+	virtual ~IGameSpawnGroupMgr() = 0;
+};
+
+class CBaseSpawnGroup : public ISpawnGroup, public IComputeWorldOriginCallback, public IGameResourceManifestLoadCompletionCallback
+{
+};
+
+class CLoadingSpawnGroup : public ILoadingSpawnGroup
+{
+};
+
+class CSpawnGroupMgrGameSystem : public IGameSpawnGroupMgr, public IGameSystem
+{
+};
```

Comment:
Specifically interested in these classes and their purpose?

Reply:
It was convenient for me to hook by final class names to count a this offset